### ISSUE TITLE
TypeError: self.text is not a function

### ISSUE THREAD

**Issue by iiminov** [ActivKonnect/angular-circles#6]
Hi there,

I am running in to this error each time I am trying to update ``text`` value on ng-circles div from my controller.

    TypeError: self.text is not a function
    at Object.<anonymous> (angular-circles.js?cdv=1878:165)
    at Scope.$get.Scope.$digest (angular.js?cdv=1878:14300)
    at Scope.$get.Scope.$apply (angular.js?cdv=1878:14571)
    at HTMLAnchorElement.<anonymous> (angular.js?cdv=1878:21571)
    at HTMLAnchorElement.jQuery.event.dispatch (jquery.js?cdv=1878:3074)
    at HTMLAnchorElement.jQuery.event.add.elemData.handle (jquery.js?cdv=1878:2750)

I have a sneaky feeling that I am doing something unreasonable here rather than this being a problem with this extension so allow me to demonstrate my use case.

OK, basically I have three ng-circles and one checkbox. When that checkbox has a value of ``false`` I don't want ng-circles to have any text. Just empty dials.

    <div data-ng-circles value="Progress" text="ProgressText"></div>

I was able to confirm the end result by modifying ``text`` property in ``ngCirclesSettingsProvider.set`` function. But I can't call upon ``ngCirclesSettingsProvider`` from controller.

So, I attempted something that I though looked similar to the examples provided:

    $scope.showHideText = function () {
        console.log('no-text: ', $scope.NoText);
        if ($scope.NoText) {
            $scope.ProgressText = $scope.Progress + '%';
        } else {
            $scope.ProgressText = "";
        }
    };

And that's when I get the error above. At first I though maybe it has something to do with the fact that my controller refers to ``['$scope',`` as ``$scope``. So I tried changing this to ``self`` as in the examples. But that didn't have any effect.

Any ideas regarding what I am doing wrong here?

**Comment by BrunoCartier** [comment]
Hey,

Thanks for using angular-circles.

Just like the original circles library, the argument given to `text` has be be a function. Giving it a string does not work.

Here is a codepen showing how to do what you are asking:
http://codepen.io/anon/pen/zGmKep

Please tell me if you have any other question.

**Comment by iiminov** [comment]
@BrunoCartier thank you very much for a prompt reply.

Now that I look at working solution one thing comes to mind. For some reason when I was looking at the examples I thought that I could shorten the solution but skipping that extra function. I guess I was wrong.

I need to be more careful in the future with these things :)

**Comment by iiminov** [comment]
Actually I just noticed something.

When ``text`` value is hidden the ``value`` attribute doesn't get reflected on the dial. In other words the circle doesn't get updated until text is enabled again.

I made a small modification to the original codepen illustrating what I noticed:
http://codepen.io/iiminov/pen/qdJRbO

I haven't yet looked into it but I think it has something to do with the way circles.js works.
